src/linter/block.go

```diff
@@ -1775,6 +1783,60 @@ func (b *BlockWalker) handleAssign(a *ir.Assign) bool {
 	return false
 }
 
+func (b *BlockWalker) handleAssignOp(o ir.Node) (res bool) {
+	var tp meta.TypesMap
+	var vr ir.Node
+
+	switch s := o.(type) {
+	case *ir.AssignPlus:
+		s.Variable.Walk(b)
+		s.Expression.Walk(b)
+		as := &ir.PlusExpr{
+			Position: s.Position,
+			Left:     s.Variable,
+			Right:    s.Expression,
+		}
+		vr = s.Variable
+		tp = solver.ExprTypeLocal(b.ctx.sc, b.r.ctx.st, as)
+	case *ir.AssignMinus:
+		s.Variable.Walk(b)
+		s.Expression.Walk(b)
+		as := &ir.MinusExpr{
+			Position: s.Position,
+			Left:     s.Variable,
+			Right:    s.Expression,
+		}
+		vr = s.Variable
+		tp = solver.ExprTypeLocal(b.ctx.sc, b.r.ctx.st, as)
+	case *ir.AssignMul:
+		s.Variable.Walk(b)
+		s.Expression.Walk(b)
+		as := &ir.MulExpr{
+			Position: s.Position,
+			Left:     s.Variable,
+			Right:    s.Expression,
+		}
+		vr = s.Variable
+		tp = solver.ExprTypeLocal(b.ctx.sc, b.r.ctx.st, as)
+	case *ir.AssignDiv:
+		s.Variable.Walk(b)
+		s.Expression.Walk(b)
+		as := &ir.DivExpr{
+			Position: s.Position,
+			Left:     s.Variable,
+			Right:    s.Expression,
+		}
+		vr = s.Variable
+		tp = solver.ExprTypeLocal(b.ctx.sc, b.r.ctx.st, as)
+	default:
+		return false
+	}
+
+	b.replaceVar(vr, tp, "assign", meta.VarAlwaysDefined)
+
+	return false
```

Comment:
I think it's easier not to return `false` and don't walk var/expr in adhoc way. It's not necessary for these kinds of nodes.